tendrl/monitoring_integration/flows/delete_resource_from_graphite/__init__.py

```diff
@@ -1,345 +1,280 @@
+    archive
+from tendrl.monitoring_integration.graphite.graphite_utils import \
+    get_data_dir_path
 
 
 class DeleteResourceFromGraphite(flows.BaseFlow):
 
     def run(self):
         super(DeleteResourceFromGraphite, self).run()
         integration_id = self.parameters.get("TendrlContext.integration_id")
-        resource_name = str(
-            self.parameters.get("Trigger.resource_name")
-        ).lower()
-        resource_type = str(
-            self.parameters.get("Trigger.resource_type")).lower()
-        self.update_graphite(
-            integration_id,
-            resource_name,
-            resource_type.lower()
+        resource_name = str(self.parameters.get("Trigger.resource_name"))
+        resource_type = str(self.parameters.get(
+            "Trigger.resource_type")).lower()
+        update_graphite(
+            integration_id, resource_name, resource_type
         )
 
-    def get_data_dir_path(self):
-        carbon_path = "/etc/tendrl/monitoring-integration/carbon.conf"
-        if not os.path.exists(carbon_path):
-            return None
-        carbon_config = ConfigParser.ConfigParser()
-        carbon_config.read(carbon_path)
-        try:
-            whisper_path = str(carbon_config.get("cache", "local_data_dir"))
-            whisper_path = os.path.join(whisper_path, "tendrl/")
-            return whisper_path
-        except KeyError:
-            return None
 
-    def delete_brick_details(
-        self,
-        integration_id,
-        resource_name,
-        whisper_path
-    ):
-        host_name = resource_name.split("|", 1)[1].split(
-            ":", 1
-        )[0].replace(".", "_")
-        brick_name = resource_name.split("|", 1)[1].split(
-            ":", 1)[1].replace("/", "|")
-        vol_name = resource_name.split("|", 1)[0]
-        archive_base_path = os.path.join(
-            whisper_path, "clusters", str(integration_id), "archive", "bricks"
-        )
-        archive_time = str(datetime.datetime.now().isoformat())
-        if not os.path.exists(archive_base_path):
-            os.makedirs(str(archive_base_path))
-        archive_path = os.path.join(
-            archive_base_path,
-            str(brick_name).replace("|", "\|") + "_" + archive_time
-        )
-        resource_path = os.path.join(
-            whisper_path, "clusters", str(integration_id), "nodes",
-            str(host_name), "bricks", str(brick_name).replace("|", "\|"))
-        brick_path = os.path.join(
-            whisper_path, "clusters", str(integration_id), "nodes",
-            str(host_name), "bricks", str(brick_name)
-        )
-        if os.path.exists(brick_path):
-            ret_val = os.system(
-                "mv " + str(resource_path) + " " + str(archive_path)
+def update_graphite(integration_id, resource_name, resource_type):
+    whisper_path = get_data_dir_path()
+    if whisper_path:
+        if resource_type == "volume":
+            delete_volume_details(
```

Comment:
We can rename the function as delete_resource_details and pass resource_type as parameter (anyway that is happening). Inside the function it would check the resource type and do the needful.

Reply:
No, we can't do this @shtripat, each function has deletion in multiple places

Reply:
Its minor thing. we can take later as well.